.pre-commit-config.yaml

```diff
@@ -98,134 +98,11 @@ repos:
-        language: pygrep
-        entry: '^(Requires|Replaces|Superseded-By):(?:(?! *( (0|[1-9][0-9]{0,3})(,|$))+$))'
-        files: '^peps/pep-\d+\.rst$'
-
-      - id: validate-created
-        name: "'Created' must be a 'DD-mmm-YYYY' date"
-        language: pygrep
-        entry: '^Created:(?:(?! +([0-2][0-9]|(3[01]))-(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)-(199[0-9]|20[0-9][0-9])$))'
-        files: '^peps/pep-\d+\.rst$'
-
-      - id: validate-python-version
-        name: "'Python-Version' must be a 'X.Y[.Z]` version"
-        language: pygrep
-        entry: '^Python-Version:(?:(?! *( [1-9]\.([0-9][0-9]?|x)(\.[1-9][0-9]?)?(,|$))+$))'
-        files: '^peps/pep-\d+\.rst$'
-
-      - id: validate-post-history
-        name: "'Post-History' must be '`DD-mmm-YYYY <Thread URL>`__, ...'"
-        language: pygrep
-        entry: '(?<=\n)Post-History:(?:(?! ?\n|((( +|\n {1,14})(([0-2][0-9]|(3[01]))-(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)-(199[0-9]|20[0-9][0-9])|`([0-2][0-9]|(3[01]))-(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)-(199[0-9]|20[0-9][0-9]) <https://((discuss\.python\.org/t/([\w\-]+/)?\d+(?:/\d+/|/?))|(mail\.python\.org/pipermail/[\w\-]+/\d{4}-[A-Za-z]+/[A-Za-z0-9]+\.html)|(mail\.python\.org/archives/list/[\w\-]+@python\.org/thread/[A-Za-z0-9]+/?(#[A-Za-z0-9]+)?))>`__)(,|(?=\n[^ ])))+\n(?=[A-Z\n]))))'
-        args: [--multiline]
-        files: '^peps/pep-\d+\.rst$'
-
-      - id: validate-resolution
-        name: "'Resolution' must be a direct thread/message URL or '`DD-mmm-YYYY <URL>`__'"
-        language: pygrep
-        entry: '(?<!\n\n)(?<=\n)Resolution: (?:(?!https://((discuss\.python\.org/t/([\w\-]+/)?\d+(/\d+)?/?)|(mail\.python\.org/pipermail/[\w\-]+/\d{4}-[A-Za-z]+/[A-Za-z0-9]+\.html)|(mail\.python\.org/archives/list/[\w\-]+@python\.org/(message|thread)/[A-Za-z0-9]+/?(#[A-Za-z0-9]+)?)))|`([0-2][0-9]|(3[01]))-(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)-(199[0-9]|20[0-9][0-9]) <(?:(?!https://((discuss\.python\.org/t/([\w\-]+/)?\d+(/\d+)?/?)|(mail\.python\.org/pipermail/[\w\-]+/\d{4}-[A-Za-z]+/[A-Za-z0-9]+\.html)|(mail\.python\.org/archives/list/[\w\-]+@python\.org/(message|thread)/[A-Za-z0-9]+/?(#[A-Za-z0-9]+)?)))>`__))\n'
-        args: ['--multiline']
-        files: '^peps/pep-\d+\.rst$'
-
-      - id: check-direct-pep-links
-        name: "Check that PEPs aren't linked directly"
-        language: pygrep
-        entry: '(dev/peps|peps\.python\.org)/pep-\d+'
-        files: '^peps/pep-\d+\.rst$'
-        exclude: '^peps/pep-(0009|0287|0676|0684|8001)\.rst$'
-
-      - id: check-direct-rfc-links
-        name: "Check that RFCs aren't linked directly"
-        language: pygrep
-        entry: '(rfc-editor\.org|ietf\.org)/[\.\-_\?\&\#\w/]*[Rr][Ff][Cc][\-_]?\d+'
-        types: ['rst']
+      # Hook to run "check-peps.py"
+      - id: "check-peps"
+        name: "Check PEPs for metadata and content enforcement"
+        entry: "python check-peps.py"
+        language: "python"
```

Comment:
The following seems to work, with `check-peps.sh` as below. `pre-commit`'s documentation is pretty poor on advanced use-cases, but it seems that it processes shebangs internally -- meaning that the nice [shebang processing](https://docs.python.org/3/using/windows.html#shebang-lines) from the Python launcher is entirely bypassed. However, `git` requires a working `bash`, so on Windows this does work.

This is the only way I've found though to avoid a needless `venv`.
 
```suggestion
        entry: "./check-peps.sh"
        language: "system"
```

```bash
#!/usr/bin/env bash
# Prefer 'python' for an active virtual environment
if hash python 2>/dev/null; then
    python ./check-peps.py
else
    ./check-peps.py
fi
```